### ISSUE TITLE
Unsupported format

### ISSUE THREAD

**Issue by PatrickvL** [Cxbx-Reloaded/Cxbx-Reloaded#849]
Many titles crash with an "unsupported format" message, which is not very difficult to prevent. However, we still haven't done this.

The main thing to consider here, is that textures that are encoded in unsupported formats, are encountered in multiple HLE patches (Register, and a few CreateTexture variants).

Instead of implementing conversions in all those places, it would be best if textures where converted only once.

The best location for this would be in an update function that should be called right before a texture is drawn with (or to, in the case of render targets).

I've done a lot of work on this, removing patches, updating active textures and whatnot to host-compatible formats, going through a cache, and much more, in my WIP Less Vertex Patching branch.

Alas, this branch has received so many changes that it is no longer feasible to rebase. But there still a lot that can be gleaned from it.

**Comment by jackchentwkh** [comment]
so how you gonna merge it without rebase? I've been waiting for this branch to be merged for a very long time.

**Comment by PatrickvL** [comment]
I stopped writing down the description of this issue too get some much needed sleep, I'll finish it later. Sorry to keep you waiting though. I wish I hadn't let it get so big either

**Comment by PatrickvL** [comment]
Note, that @Radwolfie did spend a lot of effort getting most of the non-drawing changes merged in, so thanks to him that work has been merged into master already. It's "just" the remaining rendering changes that have to be finished up

**Comment by PatrickvL** [comment]
Closing this issue, since now the new libyuv-like format conversion functions are enabled, all formats are supported
